src/cf.data.pre

```diff
@@ -4685,6 +4685,13 @@ DOC_START
 	Format codes:
 
 		%	a literal % character
+
+		byte{value}	Adds a single byte with the given value (e.g., %byte{10}
+			adds an ASCII LF character a.k.a. "new line" or "\n"). The value
+			parameter is required and must be a positive decimal integer not
+			exceeding 255. Zero-valued bytes (i.e. ASCII NUL characters) are
+			not yet supported.
```

Comment:
I call it weird since it was presented without any context nor reference to the use-case, which came up only briefly over two years ago in the community discussion.

Reply:
In planning this did you analyze the utility of making logformat accept a quoted-string as the format field when the very first character is a double-quote?
AFAIK that would allow admin to use any special characters they wanted with slash-encoding, no need for a dedicated code.

Reply:
> In planning this did you analyze the utility of making logformat accept a quoted-string as the format field when the very first character is a double-quote? AFAIK that would allow admin to use any special characters they wanted with slash-encoding, no need for a dedicated code.

I do not recall whether that specific idea was considered, but I suspect it would have been rejected. Here are a few cons:

* existing logformat configurations may start with a quote that admins want to log; relying on leading quote in a backward compatible way would require significant and largely unrelated effort
* existing logformat configurations may start with a quote that is a part of an "external" %code formatting specifications like `"%<a"`; again, relying on leading quote in a backward compatible way would require significant and largely unrelated effort
* some %code contexts cannot use quotes without additional format changes/code (e.g., see ErrorState::compileLogformatCode()).
* well-known slash-escape sequences only cover a handful of characters like `\r` and `\n`; the validity and the exact meaning of something like `\12345` is not universal and is probably not commonly recognized
* logformat configurations already use non-leading quotes and backslashes; using a leading quote would require escaping those

>  AFAIK that would allow admin to use any special characters they wanted with slash-encoding, no need for a dedicated code.

To avoid misunderstanding, I do support extending existing Squid code responsible for parsing quoted strings to support arbitrary characters. In that context, I would consider using escape sequences identical or similar to those C++23 added for [escaping characters](https://en.cppreference.com/w/cpp/language/escape) in C+ string literals.

Reply:
Overall, the choice we face is whether long-term benefits of a much more easily parsed and validated squid.conf format strings is worth the small(ish) short-term amount of pain for _some_ admins migrating config to quoted-strings. I think the long-term benefit is worth it.

Reply:
> Overall, the choice we face is whether long-term benefits of a much more easily parsed and validated squid.conf format strings is worth the small(ish) short-term amount of pain for some admins migrating config to quoted-strings. I think the long-term benefit is worth it.

I do agree that short-term pain would be acceptable if quoted-strings were the right long-term solution for logformat syntax. I disagree that migrating logformat to quoted-strings is the right long-term solution for the set of problems we want to solve. IMO, there are better options for improving logformat syntax than migrating it to quoted strings. However, this PR is not the right place to debate logformat long-term future because %byte is compatible with quoted strings and any viable alternative.

----

> Alex: relying on leading quote in a backward compatible way would require significant and largely unrelated effort

> Amos: I disagree with the "largely unrelated" part of that. If we move to using double-quoted strings instead of this %byte the effort of adding backward-compatibility would be eminently related.

Just to clarify: By "largely unrelated effort" I meant that at least 80% of the total effort to (a) "rely on a leading quote in a backward compatible way" and (b) "add support for logging of arbitrary bytes" would be spent on (a) rather than (b).